compiler/rustc_error_codes/src/error_codes/E0788.md

```diff
@@ -0,0 +1,21 @@
+A `#[no_coverage]` attribute was incorrectly placed on something that couldn't
+be covered.
```

Comment:
Hm, but then expressions can be covered, we just don't support excluding it from the coverage at that granularity. How about:

```suggestion
The `#[no_coverage]` attribute was applied to an entity which does not participate in coverage or is too granular to be excluded from the coverage report.
```

I think it would also be worthwhile to specify where the `#[no_coverage]` attribute can be applied today, something along the lines of

```
This attribute can only be applied to function, method and closure definitions.
```

perhaps?

Reply:
How about this?

> A `#[no_coverage]` attribute was applied to something which does not show up in code coverage, or is too granular to be excluded from the coverage report.
>
> For now, this attribute can only be applied to function, method, and closure definitions. In the future, it may be added to statements, blocks, and expressions, and for the time being, using this attribute in those places will just emit an `unused_attributes` lint instead of this error.

Reply:
Yeah, SGTM.